### DISCUSSION TITLE
Under which port does LinkAce run? // Setting up LinkAce with existing Nginx

### DISCUSSION THREAD

**Discussion by bytebone** [Kovah/LinkAce#539] [Support] [answered]
Hey folks,

I've been trying to get LinkAce to run on my docker server. I have Nginx running as a reverse proxy for all my apps. Each app gets a subdomain, for which I need to know the internal port that an application is running under. Until now, all apps I wanted to run have disclosed this info somewhere in their docs.
LinkAce has been set up using the Docker Advanced guide, with a modified compose.yml, where I removed the Nginx image. I've also disabled all open ports, since I don't want (and should not need) to openly expose ports when using a reverse proxy.

When running `docker ps`, port 9000 is shown as an open port. This is repeated when running `docker inspect linkace`. Therefor I wonder, what different port this app might be using. Also couldn't find any mentions of ports to be opened in the code, and also failed to find something helpful in the nginx.conf files. 

I've tried using ports 80, 8080, 8443, and 9000. But with any of them, when trying to call the subdomain, I get an error `502: Bad Gateway`, which, up to know, always meant that the port wasn't set correctly.

The `X-Forwarded-For` and ` X-Forwarded-Proto` headers should be set properly, though I don't believe that's connected to the issue.

Judging from the logs of MariaDB, Redis and LinkAce, everything is running perfectly fine. 
Maybe someone has managed to set this up and can help me with this?

Thanks.

**Comment by Kovah** [comment, marked as answer]
Hi, thanks for trying LinkAce.
You are probably using the wrong Docker image. What you are using needs nginx to access Linkace data and a specific configuration. I suggest using the linkace/linkace:simple image. Linkace is available on port 80 with this image.

**Reply by bytebone** [reply]
Yup, that sure fixed it. Thank you!

Since I only modified the version in docker compose, I am still running the entire setup from the advanced configuration. Can/Should I now get rid of Redis, since it wouldn't be a part of the basic setup? Or is it still being used, even though I am now on the simple image? 
And should I still setup the header forwarding outlined in the advanced tutorial? From what I could gather so far, both the page and the bookmarklet work perfectly fine without it, and putting the header config anywhere in my Nginx breaks access to the site. So I'd assume I no longer need these headers?

**Reply by Kovah** [reply]
Glad it works now. 😊 
> Can/Should I now get rid of Redis

Oh no, you can keep it. It improves performance and reduces disk operations.

> And should I still setup the header forwarding outlined in the advanced tutorial?

The proxy headers are a must. Otherwise you could run into issues with wrong URLs. You can take a look at the nginx.conf located in the Linkace repo.
